### DISCUSSION TITLE
Provide existing item to nested item on creation instead of greying out blank field only

### DISCUSSION THREAD

**Discussion by emahuni** [directus/directus#18989] [Feature Requests]
### Summary

Allow existing parent related item to be passed to new related item. 

The idea here is we want to be able to provide the actual item to the other when creating it as if we are editing it. Currently it just disables the input and keeps it null. 

There are many uses for this where we need to know the existing related item if we are creating it from that existing item's page. When using other custom extensions such as `directus-extension-computed` or simply use with built in 'conditions' use it's useful to know the existing item and make decisions based on it.

### Basic Example

Eg: A `house` has `o2m` relationship with `colors` that has a reverse `m2o` relationship with it. We can choose a house's `mood` (a dropdown of simple strings) but this is only available if the house has a color from which we can decide its mood.

![image](https://github.com/directus/directus/assets/7290508/d8a542a2-7370-4c2f-ab61-c522fd93efe4)

Let's say we have an existing color:

![image](https://github.com/directus/directus/assets/7290508/410e5602-20ce-4b2a-9dbb-520f733eb56d)

... and have to create a house from a color item, in the house form we can't pick decide whether to hide or show the mood dropdown because we can't tell if the color exists and is associated with the house or not since it's always null on creation.

![image](https://github.com/directus/directus/assets/7290508/006e0ccd-a2af-4d78-a65d-10ee2e1ce346)

if we had a color it was supposed to be like so (as if we are editing it):

![image](https://github.com/directus/directus/assets/7290508/b79469a4-bc70-4da6-9b15-f56d1e48a074)

### Motivation

This goes further when designing complex related models. There were 3rd stage related collections that would mean we end up repeating relationships just so that starting from any of the relationships doesn't mess up required info.

### Detailed Design

Sorry, still very new to Directus, and know I should have written an issue first, but needed this to work. I noticed that I can do the following in `app/src/interfaces/list-o2m`, am sure I need to do this with other relationships views too:

from this:

```js
function createItem() {
	currentlyEditing.value = '+';
	editsAtStart.value = { };
	newItem = true;
}
```

to this:

```js
function createItem() {
	currentlyEditing.value = '+';
        const revJunField = relationInfo.value?.reverseJunctionField?.field
	editsAtStart.value = {
            ...(!!revJunField ? {[revJunField]: values.value?.id } : {})
        };
	newItem = true;
}
```

It provided the parent item but as soon as there was an edit it was cleared. To prevent that I made another change to `src/views/private/components/drawer-item.vue`

```js
const fields = computed(() => {
	if (props.circularField) {
		return fieldsWithPermissions.value.map((field: Field) => {
			if (field.field === props.circularField) {
				// set(field, 'meta.readonly', true);
			}

			return field;
		});
	} else {
		return fieldsWithPermissions.value;
	}
});
```

that prevents the greying out of the field but works as intended... more later when I figure it out.

### Must Have:
- Must expose parent item data to relations when creating new related item from either side

### Should Have:
- 

### Could Have:
- 

### Won't Have:
- 

### Drawbacks

_

### Alternatives

_

### Adoption Strategy

add to documentation

### Unresolved Questions

Not sure how the association is saved in the API since there is another association going to be done in parent (active) item side.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
